src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a deadline to be completed. Has a description, date
+ * that the deadline is due, and can be marked as done or undone.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    public Deadline(String description, String by) {
+        this(description, by, false);
+    }
+
+    public Deadline(String description, String by, boolean isDone) {
+        super(description, 'D', isDone);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        this.by = LocalDate.parse(Arrays.stream(by.split(" ")).skip(1)
+                .collect(Collectors.joining("")), formatter);
+    }
+
+    /**
+     * Converts save string data to a Deadline object.
+     * The save string data is in the format:
+     * <p>
+     * <pre>
+     * D,"<description>","<by>"
+     * </pre>
+     * <p>
+     * 
+     * @param saveString the save string data
+     * @return the new Deadline object created from saveString
+     * @throws DukeException
+     */
+    public static Deadline fromSaveString(String saveString) throws DukeException {
+        String[] splitSaveString = saveString.split("(\",\")|(\",)|(,\")|\"");
+        if(splitSaveString.length != 3) {

```

Comment:
It would be nice if there is a spacing between if and the opening bracket.

Reply:
Thanks for pointing it out! I've fixed it (and the subsequent missing spaces) in my latest commit.